projects/samples/contests/robocup/controllers/player/robot_client.cpp

```diff
@@ -16,8 +16,13 @@
 #include <google/protobuf/io/zero_copy_stream_impl.h>
 #include <google/protobuf/text_format.h>
 #include <cmath>
+#include <iostream>
 #include <stdexcept>
 
+#if GOOGLE_PROTOBUF_VERSION < 3006001
+#define ByteSizeLong ByteSize
```

Comment:
Well. This also was my first thought but I think this is wrong. If I read the documentation correctly, `ByteSize` *could* return something different than `ByteSizeLong` (besides one being `size_t` and the other being `int`). I say this because it's not explicitly stated that these functions return the same value or that the sizes are determined the same way. That is, the documentation doesn't guarantee anything about these functions behaving the same. It's also odd, that one of them is `virtual` and the other isn't.

I personally wouldn't touch `ByteSize` at all. Or alternatively, if you really want to use it, then please use `ByteSize` with *all* protobuf versions, so that we can exclude the case that we have a different behavior with different protobuf versions (which could be very hard to debug if we ran into this)

Reply:
It seems to me that [the documentation](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message_lite#MessageLite.ByteSize) clearly states that `ByteSize` is a legacy function, the contract is pretty clear for both of them and it seems to  be a common practice to use one or the other depending on the version.

I don't think that using the legacy function with all protobuf versions is an appropriate choice... except if there is a clear reference or identified case where they are not returning the same thing, I would not use the legacy version for more recent versions of protobuf.